currency/coinmarketcap/coinmarketcap_types.go

```diff
@@ -0,0 +1,446 @@
+package coinmarketcap
+
+import "time"
+
+// Settings defines the current settings from configuration file
+type Settings struct {
+	Name        string `json:"name"`
+	Enabled     bool   `json:"enabled"`
+	Verbose     bool   `json:"verbose"`
+	APIkey      string `json:"apiKey"`
+	AccountPlan string `json:"accountPlan"`
+}
+
+// Status defines a response status JSON struct that is received with every
+// HTTP request
+type Status struct {
+	Timestamp    string `json:"timestamp"`
+	ErrorCode    int64  `json:"error_code"`
+	ErrorMessage string `json:"error_message"`
+	Elapsed      int64  `json:"elapsed"`
+	CreditCount  int64  `json:"credit_count"`
+}
+
+// CryptoCurrencyInfo defines cryptocurrency information
+type CryptoCurrencyInfo map[string]struct {
+	ID       int         `json:"id"`
+	Name     string      `json:"name"`
+	Symbol   string      `json:"symbol"`
+	Category string      `json:"category"`
+	Slug     string      `json:"slug"`
+	Logo     string      `json:"logo"`
+	Tags     []string    `json:"tags"`
+	Platform interface{} `json:"platform"`
+	Urls     struct {
+		Website      []string      `json:"website"`
+		Explorer     []string      `json:"explorer"`
+		SourceCode   []string      `json:"source_code"`
+		MessageBoard []string      `json:"message_board"`
+		Chat         []interface{} `json:"chat"`
+		Announcement []interface{} `json:"announcement"`
+		Reddit       []string      `json:"reddit"`
+		Twitter      []string      `json:"twitter"`
+	} `json:"urls"`
+}
+
+// CryptoCurrencyMap defines a cryptocurrency struct
+type CryptoCurrencyMap struct {
+	ID                  int         `json:"id"`
+	Name                string      `json:"name"`
+	Symbol              string      `json:"symbol"`
+	Slug                string      `json:"slug"`
+	IsActive            int         `json:"is_active"`
+	FirstHistoricalData time.Time   `json:"first_historical_data"`
+	LastHistoricalData  time.Time   `json:"last_historical_data"`
+	Platform            interface{} `json:"platform"`
+}
+
+// CryptocurrencyHistoricalListings defines a historical listing data
+type CryptocurrencyHistoricalListings struct {
+	ID                int       `json:"id"`
+	Name              string    `json:"name"`
+	Symbol            string    `json:"symbol"`
+	Slug              string    `json:"slug"`
+	CmcRank           int       `json:"cmc_rank"`
+	NumMarketPairs    int       `json:"num_market_pairs"`
+	CirculatingSupply float64   `json:"circulating_supply"`
+	TotalSupply       float64   `json:"total_supply"`
+	MaxSupply         float64   `json:"max_supply"`
+	LastUpdated       time.Time `json:"last_updated"`
+	Quote             struct {
+		USD struct {
```

Comment:
If you extract this struct to its own type, you can reuse it many times. Even more if you have an omitempty `LastUpdated` field.